### PR TITLE
feat: add /shipwright:unblock command for HITL triage (HSR-2.5)

### PR DESCRIPTION
**Pull request by dmcaulay**
## Summary
- Adds `/shipwright:unblock` — a human-facing, interactive triage command that discovers `status:'blocked'` tasks (`GET /tasks?state=blocked`) and `blocked:true` PRs (`GET /prs?blocked=true`) across the caller's repo scope.
- Infers which pipeline phase (dev-task, patch, deploy, or spin-detection) raised each escalation by matching `blockedReason` text against the known prefixes from each phase's escalation sites, and walks the human through retry, redirect, or abandon per item.
- Retry correctly branches three ways: task-with-PR → `status:'pr_open'` (avoids a duplicate PR from dev-task); no-PR task → `POST /tasks/:id/release` (agent tokens can't PATCH `status:'pending'` directly) followed by clearing `blockedReason`/`blockedAt`; PR-only record → `PATCH /prs/:id` with `blocked:false`. Spin-detected blocks additionally reset `skipCount` via `POST /tasks/:id/skip/reset`. Abandon sets `status:'cancelled'` while explicitly preserving `blockedReason` as a historical note.
- Exempt from the Candidate Selection Contract (`plugins/shipwright/CLAUDE.md`) by design — it self-discovers because it's a standalone human tool, not a `shipwright-loop`-dispatched phase.

## Acceptance Criteria
| # | Criterion | Status |
|---|-----------|--------|
| 1 | Discovers all status:'blocked' tasks and blocked:true PRs across target repo(s) | MET |
| 2 | Correctly infers origin phase for each blockedReason pattern | MET |
| 3 | Retry restores task.pr-having tasks to pr_open (not pending), PR-only records to blocked:false | MET |
| 4 | Abandon sets status:'cancelled' with blockedReason preserved | MET |
| 5 | New unblock.content.test.ts covering discovery, phase-inference, retry/redirect/abandon | MET |

## Test Plan
- `unblock.content.test.ts` — 26 tests covering discovery, phase-inference table, all three retry destinations, redirect, abandon, and edge cases (nothing-to-triage, unknown blockedReason).
- `task ci` — lint, check-strings, typecheck, check-config-docs, check-version-sync, coverage gate (81.35%/82.21% vs 80/80 threshold) all pass. One pre-existing unrelated failure (`agent/src/claude.unit.test.ts`'s `extraAllowedTools` test) reproduces identically on clean main with `agent/` untouched by this diff — a known environment gap, not caused by this change.
- [x] Pre-commit checks passing

Generated with [Claude Code](https://claude.com/claude-code)

### CODE REVIEW

File: plugins/shipwright/commands/unblock.md [hunk 1]
```diff
@@ -0,0 +1,271 @@
...
+| **patch** | `merge-conflict` (resolution blocked) |
+| **patch** | `second-round disagreement` (reviewer vs. automated fix, escalated to HITL) |
+| **patch** | `review-finding fix blocked` |
+| **patch** | `CI-fix blocked` |
+| **deploy** | `Post-merge CI failed` |
+| **deploy** | `Deploy stage failed` |
+| **deploy** | `canary_blocked` |
+| **deploy** | `Pipeline timeout` |
+| **deploy** | `Canary failed after deploy` |
+| **spin-detection** (`recordSkip`) | starts with `Auto-blocked after` — exact format is `` Auto-blocked after ${N} consecutive skips (dispatched but found nothing to do) `` |
```
Comment (dodizzle) [inline plugins/shipwright/commands/unblock.md]: Verified against current `main`: this table is missing two real cases from the phases it claims to exhaustively cover (AC2 says this is MET).

1. `deploy.md:483` sets a PR-only `blockedReason: "Post-merge CI still pending after 10 minutes — marking deployed, check manually"` when the post-merge CI budget times out with runs still pending — doesn't match `Post-merge CI failed` above, so it falls into "unknown origin phase" despite coming from `deploy`.
2. `pull-request-service.ts:618` sets a PR-only `blockedReason` of the form `` Auto-blocked after ${N} consecutive patch cycles hitting the same CI failure (${signature}) `` — this *does* match the `Auto-blocked after` prefix here, but it's a distinct counter (`consecutiveCiFailureCount` on the PR, not `skipCount` on a task). Step 6d's reset (`/tasks/:id/skip/reset`) doesn't apply (task-only, and wrong counter anyway) — there's no reset endpoint for `consecutiveCiFailureCount`, so a retry via 6c clears the symptom but the streak counter stays elevated and one more matching CI failure re-blocks the PR immediately.

Suggest adding a `deploy` row for the pending-timeout variant and splitting spin-detection into the task-side `recordSkip` case vs. this PR-side CI-failure-streak case, with a note that the latter has no dedicated reset today.

**Review by dodizzle — COMMENTED** [review]
Verdict: COMMENT — Phase-inference table (Step 4) misses two real blockedReason variants (deploy's post-merge-CI-pending timeout, and patch's per-PR CI-failure-streak spin-detection, which is mislabeled under the wrong recordSkip/skipCount reset). Everything else checks out against current main source.

**Review by dmcaulay — COMMENTED** [review]
Verdict: COMMENT — Step 4's phase-inference table mislabels the `consecutive skips` spin-detection pattern as task-only; PRs have their own identical skipCount/recordSkip/resetSkip and a live /prs/:id/skip/reset endpoint, so a PR-only blocked record with this reason gets no reset guidance and re-blocks immediately. Everything else — including the prior review round's fix (missing deploy pending-timeout row, CI-failure-streak split) — checks out against current source.

File: plugins/shipwright/commands/unblock.md [hunk 2]
```diff
@@ -0,0 +1,283 @@
...
+| **patch** | `second-round disagreement` (reviewer vs. automated fix, escalated to HITL) |
+| **patch** | `review-finding fix blocked` |
+| **patch** | `CI-fix blocked` |
+| **deploy** | `Post-merge CI failed` |
+| **deploy** | `Deploy stage failed` |
+| **deploy** | `canary_blocked` |
+| **deploy** | `Pipeline timeout` |
+| **deploy** | `Canary failed after deploy` |
+| **deploy** | `Post-merge CI still pending after 10 minutes` (post-merge CI budget timed out with runs still pending — PR-only, set by `deploy.md`'s Step 5a) |
+| **spin-detection — task skip (`recordSkip`)** | starts with `Auto-blocked after` and contains `consecutive skips` — exact format is `` Auto-blocked after ${N} consecutive skips (dispatched but found nothing to do) `` — task-only, tracked via `skipCount` |
```
Comment (dmcaulay) [inline plugins/shipwright/commands/unblock.md]: This row (and Step 6d below) labels this pattern "task-only, tracked via `skipCount`" — but `PullRequest` has its own `skipCount`/`lastSkippedAt` (task-store/prisma/schema.prisma:192-193) and `PullRequestService.recordSkip()` (task-store/src/pull-request-service.ts:805) produces the *identical* blockedReason string on a PR-only record. `loop-orchestrator.ts` calls `recordSkip("pr", ...)` on PR candidates too, and `POST /prs/:id/skip/reset` exists (task-store/src/routes/prs.ts:301). A human following this doc on a PR-only record with this exact reason is told the reset step doesn't apply, leaves `skipCount` elevated, and the PR re-blocks on the next skip — worth adding a PR-side branch to 6d that calls `/prs/:id/skip/reset`.

File: plugins/shipwright/commands/unblock.md [hunk 3]
```diff
@@ -0,0 +1,296 @@
...
+|---|---|
+| **dev-task** | `implementation_blocked_after_model_escalation` |
+| **dev-task** | `requirements_not_met` |
+| **dev-task** | `pr_creation_failed` |
+| **dev-task** | `ci_max_retries_exhausted` |
+| **patch** | `merge-conflict` (resolution blocked) |
+| **patch** | `second-round disagreement` (reviewer vs. automated fix, escalated to HITL) |
+| **patch** | `review-finding fix blocked` |
+| **patch** | `CI-fix blocked` |
+| **deploy** | `Post-merge CI failed` |
```
Comment (dmcaulay) [inline plugins/shipwright/commands/unblock.md]: This table (and Step 4's lead-in at line 97) instructs parsing `.blockedReason` on tasks to match these 5 deploy patterns. But `deploy.md` (lines 452, 656, 685, 715, 790) always writes these strings to the task's `note` field when a task is linked (`$TASK_ID` set) — `blockedReason` is only populated in the PR-only branch. For a task-linked deploy escalation, `task.blockedReason` stays null, so this table never matches: Step 5 shows an empty "Blocked reason:" line and Step 4 falls through to "unknown origin phase" instead of correctly attributing it to deploy. Suggest also checking `task.note` for these 5 patterns when `task.blockedReason` is empty.

**Review by dmcaulay — COMMENTED** [review]
Verdict: COMMENT — The prior round's finding (PR-side skipCount mislabeled task-only) is correctly fixed. New finding: Step 4's phase-inference table parses `.blockedReason` for all 5 deploy-phase patterns, but `deploy.md` always writes these strings to the task's `note` field (not `blockedReason`) when a task is linked — only the PR-only branch populates `blockedReason`. So every task-linked deploy escalation will show as "unknown origin phase" with an empty blocked-reason line, contradicting AC2. Everything else (dedup logic, retry-destination branching, Candidate Selection Contract exemption, task-store API semantics) checks out against current source.

**Review by dodizzle — APPROVED** [review]
Verdict: APPROVE — All three review rounds' findings (missing deploy pending-timeout row, PR-side skipCount mislabeled task-only, task.note fallback for deploy blockedReason) are verified fixed against current source. No new issues found.
